### PR TITLE
fix!: drop python 2.7 support

### LINKED ISSUE #102: Drop Python 2.7 support.
This would be semver major, including moving to depend on `google-api-core 2.0.0` (see: https://github.com/googleapis/python-api-core/issues/210).

Clients which cannot move forward (e.g, `google-cloud-storage`) already pin `google.cloud.core < 2.0.0dev` directly.

Extra cleanups:

- [x] Drop `mock` wrapper (use `from unittest import mock` directly).
- [x] Drop `six`.
- [x] Drop `u"` prefixes for text strings.
- [x] Drop `pytz` (as in https://github.com/googleapis/python-api-core/issues/73).

### PR DESCRIPTION
**Pull request by tseaver**
- Drop use of 'mock' shim.
- Drop use of 'six'.
- Drop explicit 'u"' prefixes.
- Drop 'pytz'.

Release-As: 2.0.0b1

Closes #102.

### CODE REVIEW

File: google/cloud/_testing.py
Comment (plamut) [inline google/cloud/_testing.py:122]: Since dropping Python 2, we could also drop the entire `next()` method and just use `__next__()`.

File: tests/unit/test__helpers.py
Comment (plamut) [inline tests/unit/test__helpers.py:476]: There are also a few remaining occurrences of these prefixes in `docs/conf.py`.

**Review by plamut — COMMENTED** [review]
Generally looks good to me, just a few bits remaining - things from the comments and a few mentions of "Python 2" and "2.7" in the docs that are now obsolete, e.g in `CONTRIBUTING.rst` and at the end of README.

File: google/cloud/version.py
Comment (tseaver) [reply google/cloud/version.py:15]: As I said in https://github.com/googleapis/python-api-core/pull/212, I think a beta is sensible for a semver-major breaking change.  I'll let @crwilcox and @busunkim96 chime in on that.

File: tests/unit/test__helpers.py
Comment (tseaver) [reply tests/unit/test__helpers.py:476]: Templated file.

File: tests/unit/test__helpers.py
Comment (tseaver) [reply tests/unit/test__helpers.py:16]: 4a7e7eb

File: google/cloud/_testing.py
Comment (tseaver) [reply google/cloud/_testing.py]: 9292b7b

File: tests/unit/test__helpers.py
Comment (tseaver) [reply tests/unit/test__helpers.py:476]: Update upstream pending: https://github.com/googleapis/synthtool/pull/1119

**Review by plamut — COMMENTED** [review]
Also update the "Deprecated Python Versions" section in README like in other libraries?

Looks good otherwise, IMO.

**Comment by tseaver** [comment]
@plamut Thanks for the review!  fca858a updates the README to document last version supporting Python 2.7.

**Comment by tseaver** [comment]
Marked as `do not merge` pending discussion over the versioning policy next week with @busunkim96, @crwilcox, et al.

**Comment by plamut** [comment]
@tseaver https://github.com/googleapis/python-cloud-core/commit/12014e293a28a6a0e4ec25363859623c15323da5 has not been pushed yet? I can still see references to Python 2.7 in CONTRIBUTING when I pull the PR branch locally.

**Comment by tseaver** [comment]
@plamut Owlbot is running in automated fashion, pushing changes to PRs.  I guess we have to apply the changes to `CONTRIBUTING.rst` in `owlbot.py` via `s.replace()` (or else exclude it -- see 85775cc).

**Review by plamut — APPROVED** [review]

File: CONTRIBUTING.rst
Comment (parthea) [inline CONTRIBUTING.rst:154]: I'm going to open an issue in the synthtool repo to request that we update the templated `CONTRIBUTING.rst`

Comment (parthea) [reply CONTRIBUTING.rst:154]: I've updated https://github.com/googleapis/synthtool/issues/1118 to track this issue.

Comment (tseaver) [reply CONTRIBUTING.rst:154]: https://github.com/googleapis/synthtool/issues/1126

Comment (parthea) [reply CONTRIBUTING.rst:154]: Approved and merged googleapis/synthtool#1127. Owlbot should automatically run on this PR.

Comment (tseaver) [reply CONTRIBUTING.rst]: @parthea Apparently not: fa4dc90 still puts the `2.7` bits back.

Comment (parthea) [reply CONTRIBUTING.rst]: My mistake, it won't run automatically because the post processor SHA in this PR is not the latest one. Master has [this SHA](https://github.com/googleapis/python-cloud-core/blob/master/.github/.OwlBot.lock.yaml#L3), but the latest SHA is [this one](https://github.com/googleapis/python-cloud-core/blob/2058556e8c8489d0776b743ae2f222e0181f276d/.github/.OwlBot.lock.yaml#L3).

We have 2 options, either update the `.OwlBot.lock.yaml` in this PR to use the latest SHA `b6169fc6a5207b11800a7c002d0c5c2bc6d82697185ca12e666f44031468cfcd` or merge #105 and rebase this branch. The former seems like a best option as this is the PR that actually drops python 2.7 support.

Comment (parthea) [reply CONTRIBUTING.rst]: IOW, Owl bot should run automatically once `.OwlBot.lock.yaml` is updated in this PR with SHA `b6169fc6a5207b11800a7c002d0c5c2bc6d82697185ca12e666f44031468cfcd`

**Comment by tseaver** [comment]
Ugh, OwlBot is restoring the `2.7` reference in the templated `scripts/readme-gen/templates/install_deps.tmpl.rst'.  See https://github.com/googleapis/synthtool/pull/1173.

**Pull request merged**
